app/modules/asset_groups/schemas.py

```diff
@@ -42,6 +54,48 @@ class Meta(BaseAssetGroupSightingSchema.Meta):
         dump_only = BaseAssetGroupSightingSchema.Meta.dump_only
 
 
+class AssetGroupSightingAsSightingSchema(BaseAssetGroupSightingSchema):
+    """
+    In order for the frontend to render an AGS with the same code that renders a
+    sighting, we have to pop out all the fields in AGS.config into the top-level
+    of the schema. This is done with AssetGroupSighting.config_field_getter,
+    which creates a getter for a config field of a given name. We can add more
+    fields using the pattern below.
+    """
+
+    assets = base_fields.Function(AssetGroupSighting.get_assets)
+    completion = base_fields.Function(AssetGroupSighting.get_completion)
+
+    # Note: these config_field_getter vars should conform to SIGHTING_FIELDS_IN_AGS_CONFIG
+    # at the top of this file
+    decimalLatitude = base_fields.Function(
+        AssetGroupSighting.config_field_getter('decimalLatitude')
+    )
+    decimalLongitude = base_fields.Function(
+        AssetGroupSighting.config_field_getter('decimalLongitude')
+    )
+    locationId = base_fields.Function(
+        AssetGroupSighting.config_field_getter('locationId')
+    )
+    startTime = base_fields.Function(AssetGroupSighting.config_field_getter('startTime'))
+    encounters = base_fields.Function(
+        AssetGroupSighting.config_field_getter('encounters')
+    )
+
+    class Meta(BaseAssetGroupSightingSchema.Meta):
```

Comment:
I'm getting an exception:

```
app/modules/asset_groups/schemas.py:57: in <module>
    class AssetGroupSightingAsSightingSchema(BaseAssetGroupSightingSchema):
/usr/local/lib/python3.9/site-packages/marshmallow/schema.py:106: in __new__
    klass.opts = klass.OPTIONS_CLASS(meta)
/usr/local/lib/python3.9/site-packages/flask_marshmallow/sqla.py:33: in __init__
    super(SchemaOpts, self).__init__(meta)
/usr/local/lib/python3.9/site-packages/marshmallow_sqlalchemy/schema.py:38: in __init__
    super(ModelSchemaOpts, self).__init__(meta)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
                         
self = <flask_marshmallow.sqla.SchemaOpts object at 0x7f80c9413940>
meta = <class 'app.modules.asset_groups.schemas.AssetGroupSightingAsSightingSchema.Meta'>
                         
    def __init__(self, meta):
        self.fields = getattr(meta, 'fields', ())
        if not isinstance(self.fields, (list, tuple)):
            raise ValueError("`fields` option must be a list or tuple.")
        self.additional = getattr(meta, 'additional', ())
        if not isinstance(self.additional, (list, tuple)):
            raise ValueError("`additional` option must be a list or tuple.")
        if self.fields and self.additional:
>           raise ValueError("Cannot set both `fields` and `additional` options"
                            " for the same Schema.")
E           ValueError: Cannot set both `fields` and `additional` options for the same Schema.

/usr/local/lib/python3.9/site-packages/marshmallow/schema.py:193: ValueError
```

We need to not inherit from `BaseAssetGroupSightingSchema.Meta` or add `field = ()` because `BaseAssetGroupSightingSchema` defines `field` and you can't define both `fields` and `additional`